README.md

```diff
@@ -21,162 +23,154 @@ repositories {
+            Gson().fromJson(str, Endpoint::class.java)
+    }
 }
 
-// use the custom data class
+// Use the custom data class
 val defaultEndpoints =
     listOf(
         Endpoint("Live", "some.live.endpoint.org", 2, false),
         Endpoint("Dev", "some.dev.endpoint.org", 6, true)
     )
 
-// init the library
+// Init the library
 val endpointPicker = envPicker(
-    Config(
-        "endpointsPicker", // used as sharedPrefs key
-        "Choose Endpoint", // displayed as fragment title
-        EntryDescription(
-            // these define each field's type and name that will be displayed in the UI
-            listOf(
-                FieldDescription("URL", FieldType.String),
-                FieldDescription("Retry Count", FieldType.Int),
-                FieldDescription("Allow HTTP", FieldType.Boolean),
-            ),
-            // how to create an Endpoint from user inputs
-            { name, fields ->
-                Endpoint(name, fields[0] as String, fields[1] as Int, fields[2] as Boolean)
-            },
-            // how to serialize it
-            // we use Gson here, but you can use any serialization method you like
-            { entry -> Gson().toJson(entry) },
-            // and how to deserialize it
-            { str -> Gson().fromJson(str, Endpoint::class.java) }
-        ),
-        // which Endpoints should be available per default?
-        defaultEndpoints,
-        // which endpoint should be active initially?
-        defaultEndpoints[0]
-    ),
-    context
+    key = "endpointsPicker", // Used as sharedPrefs key
+    fragmentTitle = "Choose Endpoint", // Displayed as title of picker UI
+    // Which Endpoints should be available per default?
+    defaultEntries = defaultEndpoints,
+    // Which endpoint should be active initially?
+    defaultActiveEntry = defaultEndpoints[0],
+    // Optional: Use your custom serializer
+    customSerializer = Endpoint.Serializer(),
+    context = context,
 )
 
-// accessing the entry's fields now works with the custom field names
+// Accessing the current entry now returns an instance of your custom data class
 val currentlyActiveEndpointUrl =
-    endpointPicker.getActiveEntry(context)?.url
+    endpointPicker.getActiveEntry(context).url
 val currentlyActiveEndpointRetryCount =
-    endpointPicker.getActiveEntry(context)?.retryCount
+    endpointPicker.getActiveEntry(context).retryCount
 ```
 
-### Supported FieldTypes
+### Multiple instances
+
+You can easily define multiple instances of `EnvPicker`, each managing another aspect of your environment. Just make sure to provide unique `key`s in each call to `envPicker`!
+
+### Supported field types
 
 The currently supported types for custom data class fields are:
 
 - String
 - Int
 - Boolean
 
+## Obfuscation
+
+In case you enable obfuscation through R8 and you are using a custom data class, the following needs
+to be included in your rules:
+
+```
+-keep class <customClassQualifiedName> { *; }
```

Comment:
Good and important point - Does it make sense to enforce from the code side that you have to implement an interface specified in the default ProGuard config, something like `Serializable`?

Reply:
Since this is the client app's proguard config, the library can't define any default rules for the data class. The app needs to prevent obfuscation in some way. However, I believe the standard config about `Serializable` is not enough, since we need to keep the annotations on the data class constructor parameters.